build-system/app.js

```diff
@@ -832,13 +832,10 @@ app.use([cloudflareDataDir], function fakeCors(req, res, next) {
  */
 app.get([fakeAdNetworkDataDir + '/*', cloudflareDataDir + '/*'], (req, res) => {
   let filePath = req.path;
-  let unwrap = false;
-  if (req.path.endsWith('.html')) {
-    filePath = req.path.slice(0,-5);
-    unwrap = true;
-  }
+  let unwrap = !req.path.endsWith('.html');
   filePath = pc.cwd() + filePath;
   fs.readFileAsync(filePath).then(file => {
+    res.setHeader('X-AmpAdRender', 'nameframe');
```

Comment:
(Copying comment over from outdated code, with slight edits for clarity)
@lannka Re: your previous question, it turns out that the problem was here. "unwrap" was implemented backwards. If the creative is a .json file, then it should be unwrapped (signature and HTML extracted), and if it's a .html file then that is not needed (and the extension should NOT be stripped off). But "unwrap" was done the opposite way. My other main change here is the line that sets the X-AmpAdRender header, which removes the need for the change to amp-a4a.js which you weren't a fan of. Now, https://github.com/ampproject/amphtml/blob/49951ce9047a634fdcfc04e49fa1ca3dcff92e9a/extensions/amp-a4a/0.1/amp-a4a.js#L611 sets method correctly, and we don't hit the error on line 615.